File: shlee/BOJ/Gold/AC.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.ArrayDeque;
import java.util.ArrayList;
import java.util.Collections;
import java.util.Deque;
import java.util.List;

class Main {

  public static void main(String[] args) throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    int T = Integer.parseInt(br.readLine()); // 케이스 개수

    while (T-- > 0) {
      String AC = br.readLine(); // 수행할 연산들
      int count = Integer.parseInt(br.readLine()); // 배열에 들어있는 수의 개수
      String arrString = br.readLine(); // 배열을 나타내는 문자열
      String[] split = arrString.substring(1, arrString.length() - 1).split(","); // []떼고 ","를 기준으로 split
      Deque<Integer> deque = new ArrayDeque<>(); // 입력받은 정수들을 저장할 데크
      boolean isReverse = false; // 배열이 뒤집어졌는지 체크
      boolean isEmpty = false; // 에러 발생여부 체크

      for (int i = 0; i < count; i++) {
        deque.add(Integer.valueOf(split[i]));
      }

      for (int i = 0; i < AC.length(); i++) {
        if (AC.charAt(i) == 'R') {
          isReverse = !isReverse;
        } else {
          if (deque.isEmpty()) {
            isEmpty = true;
          }
          if (isReverse) {
            if (!isEmpty) deque.removeLast();
          } else {
            if (!isEmpty) deque.removeFirst();
          }
        }
      }

      if (isEmpty) {
        bw.write("error\n");
      } else {
        StringBuilder sb = new StringBuilder("[");
        while (!deque.isEmpty()) {
          sb.append(isReverse ? deque.removeLast() : deque.removeFirst());
          if (!deque.isEmpty()) sb.append(",");
        }
        sb.append("]\n");
        bw.write(String.valueOf(sb));
      }
    }

    br.close();
    bw.flush();
    bw.close();
  }
}

File: syJoh/BOJ/AC.py
from collections import deque
import sys
input = sys.stdin.readline
t = int(input())
while t>0:
    p = input().strip() #수행할 함수
    n = int(input().strip()) #배열 개수
    arr = input().strip()
    q = deque(arr[1:-1].split(","))
    if n==0:
        q = []
    rcount = 0
    flag = True
    for i in p:
        if i == "R":
          rcount += 1
        elif i == "D":
          if len(q) == 0:
            flag = False
            break
          else:
            if rcount%2==0:
              q.popleft()
            else:
              q.pop()
    if flag:
        if rcount%2!=0 :
            q.reverse()
        print("["+",".join(list(q))+"]")
    else:
        print("error")
    t-=1
"""
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int t = Integer.parseInt(br.readLine());
        while(t>0){
            String[] pList = br.readLine().split("");//함수
            int n = Integer.parseInt(br.readLine());
            Deque<Integer> q = new LinkedList<>();
            String str = br.readLine(); //배열 형태의 정수
            String[] arr = {};
            if(n!=0){
                arr = str.substring(1, str.length()-1).split(",");
            }
            for(int i=0; i<arr.length; i++){
                q.add(Integer.parseInt(arr[i]));
            }
            boolean isRotated = false;
            boolean flag = true;
            for(String p : pList){
                if(p.equals("R")){
                    isRotated = !isRotated;
                }else if(p.equals("D")){
                    if(q.isEmpty()){
                        flag = false;
                        break;
                    }
                    if(isRotated){
                        q.pollLast();
                    }else{
                        q.pollFirst();
                    }
                }
            }
            if(flag){
                if(isRotated){ //홀수일 경우 회전
                    List<Integer> list = new ArrayList<>(q);
                    Collections.reverse(list);
                    q = new LinkedList<>(list);
                }
                sb.append("[");
                while(q.size()>1){
                    sb.append(q.poll()+",");
                }
                if(q.peek()!=null) {
                    sb.append(q.poll());
                }
                sb.append("]\n");
            }else{
                sb.append("error\n");
            }
            t--;
        }
        br.close();
        System.out.print(sb);
    }
}
"""
